File: Java/src/main/java/org/codecop/dependencies/a/Discount.java
package org.codecop.dependencies.a;

public class Discount {

    private final MarketingCampaign marketingCampaign;

    public Discount(MarketingCampaign marketingCampaign1) {
        this.marketingCampaign = marketingCampaign1;
    }

    public Money discountFor(Money netPrice) {
        if (marketingCampaign.isCrazySalesDay()) {
            return netPrice.reduceBy(15);
        }
        if (netPrice.moreThan(Money.ONE_THOUSAND)) {
            return netPrice.reduceBy(10);
        }
        if (netPrice.moreThan(Money.ONE_HUNDRED) && marketingCampaign.isActive()) {
            return netPrice.reduceBy(5);
        }
        return netPrice;
    }
}

File: PHP/src/A/Discount.php
<?php

namespace Org\Codecop\Dependencies\A;

class Discount {

    private $marketingCampaign;

    public function __construct() {
        $this->marketingCampaign = new MarketingCampaign();
    }

    public function discountFor(Money $netPrice): Money {
        if ($this->marketingCampaign->isCrazySalesDay()) {
            return $netPrice->reduceBy(15);
        }
        if ($netPrice->moreThan(Money::ONE_THOUSAND)) {
            return $netPrice->reduceBy(10);
        }
        if ($netPrice->moreThan(Money::ONE_HUNDRED) && $this->marketingCampaign->isActive()) {
            return $netPrice->reduceBy(5);
        }
        return $netPrice;
    }
}
